### ISSUE TITLE
GLIGEN for SD v2 or XL

### ISSUE THREAD

**Issue by TonyLianLong** [gligen/GLIGEN#46]
The released weights are based on SD v1.4. Are there any plans for v2 and XL support?

**Comment by zengjie617789** [comment]
you should train it by yourself and what sad is  it shoulbe try every time if you want to change your base model which cost much time.

**Comment by zjysteven** [comment]
Hi @TonyLianLong, would you be interested in collaborating on extending GLIGEN to v2 and XL? I tried some training on SDXL but unfortunately haven't had luck in getting decent results.

**Comment by Acephalia** [comment]
> Hi @TonyLianLong, would you be interested in collaborating on extending GLIGEN to v2 and XL? I tried some training on SDXL but unfortunately haven't had luck in getting decent results.

@zjysteven I had a look into this yesterday myself and quite keen to see if we can make any progress.

**Comment by zjysteven** [comment]
@Acephalia I've made some attempts earlier but didn't have any luck. I've also seen people struggling to reproduce the training even with the original code on SD1.5, so finally I decided to stop trying. There are some other training-free controlled generation method like BoxDiff, which I ended up using.

**Comment by Acephalia** [comment]
Yeah that seems to track from what everyone else has said too the paper seems a little clunky and I was having real trouble even understanding the training bits. Thanks for taking the time to reply.

**Comment by rezponze** [comment]
Does this mean GLIGEN for SDXL is a dead end?

**Comment by zcfrank1st** [comment]
any update？

**Comment by ynie** [comment]
+ 1

**Comment by mhillebrand** [comment]
And what about SD3?

**Comment by camoody1** [comment]
> @Acephalia I've made some attempts earlier but didn't have any luck. I've also seen people struggling to reproduce the training even with the original code on SD1.5, so finally I decided to stop trying. There are some other training-free controlled generation method like BoxDiff, which I ended up using.

What is BoxDiff? I've never heard about this. Would you mind sharing a workflow that makes use of it?

**Comment by TonyLianLong** [comment]
I just built a codebase called [IGLIGEN](https://github.com/TonyLianLong/igligen) that I use to train GLIGEN on SDv1.5/2.1. It also supports ModelScope (text-to-video generation). It is trained with SA-1B dataset which is only ~300GB after preprocessing and has 11M images. It is more "modern" (i.e., it is based on diffusers training script) and supports flash attention. The dataset, preprocessing script, and training script are included in the repo. Feel free to contact me for suggestions for this repo.

Repo: https://github.com/TonyLianLong/igligen
